### DISCUSSION TITLE
Angular gateway-api build is failing - Missing bff-gateway.openapi.json file

### DISCUSSION THREAD

**Discussion by saravanad** [ch4mpy/spring-addons#109] [Q&A] [answered]
Hi @ch4mpy ,

I cloned the project and wanted to try out the BFF usecase. I tried to do a npm i on the angular project and this is the error message that i got. I don't have much expertise on the angular side. Anything that i am doing it incorrectly. I cloned the master branch.

<img width="846" alt="image" src="https://user-images.githubusercontent.com/3892196/232237404-ca72adbd-c9d2-4cbf-abd6-06557463adfd.png">

Any pointers will be really helpful

Thanks,
Saravana

**Comment by ch4mpy** [comment, marked as answer]
This is actually a bug (sorry about that). I should have either:
- not excluded the OpenAPI from git tracking (which I just reverted, please `git pull` to fetch it)
- instructed to build the projects with openapi profiles: `mvn install -P openapi-webflux` in gateway and `mvn install -P openapi-webmvc` in greeting-api (springdoc-api dependencies are different for servlet and reactive apps), but even that was broken :/

It should be fixed, now (both: the required files are in tracked by git and you should be able to refresh it with maven profiles). Please re-open if it's not the case.

**Reply by saravanad** [reply]
Thanks @ch4mpy  for the quick Response. Really appreciate.

Now i am able to build the angular project. When i run ng serve command, i am getting the following Issue

<img width="857" alt="image" src="https://user-images.githubusercontent.com/3892196/232246177-3eea7423-8114-4023-a957-0d4dafc5e83d.png">


Also when i try to view the angular app in the browser, i get the following 

<img width="660" alt="image" src="https://user-images.githubusercontent.com/3892196/232246215-98b72e64-8791-47d8-a9eb-6c7b19f93135.png">


Any pointers on this will be really helpful.

**Reply by saravanad** [reply]
Have fixed the compliation error. now the Angular UI is working

**Reply by ch4mpy** [reply]
I have just fixed the OpenAPI spec (the strings in DTOs from APIs should be neither nullable nor empty). I also removed the secure headers on gateway for the Angular app to work without https (I usually always have SSL enabled, even on my dev env and hadn't noticed it wasn't working without SSL). Please pull again.

Please be sure to:
- start the Angular app with `npm run start` (or to serve with `development` configuration)
- access Angular app through the gateway (http://localhost:8080 and not http://localhost:4200 or requests to APIs will fail)

**Reply by saravanad** [reply]
Thanks @ch4mpy for the update. I pulled the changes and followed the steps as you have mentioned.

when i hit http://localhost:8080 - i get blank page with the following error in the network tab. The JS files are not being loaded due to CSP.

<img width="747" alt="image" src="https://user-images.githubusercontent.com/3892196/232333136-f13d503a-06a3-4172-b0fa-2b0252815184.png">

Any pointers on clearing this issue will be really helpful. Thanks in advance

**Reply by saravanad** [reply]
One thing that we observed is that 
When we try http://localhost:4200/ui/ in the browser, the JS files are loading without any issues.

If we try http://localhost:8080 - then having the above mentioned CSP JS loading issue

**Reply by ch4mpy** [reply]
@saravanad it really looks like your `spring.cloud.gateway.default-filters` still contain `SecureHeaders` (gateway application.yaml), which it shouldn't when SSL is disabled. I pushed an update yesterday where `SecureHeaders` is present in `ssl` profile only. Are you sure you have the latest version?

**Reply by saravanad** [reply]
Thanks @ch4mpy It worked. Had overlooked the secureHeaders.
